plugins/repository-gcs/src/main/java/org/opensearch/repositories/gcs/GoogleApplicationDefaultCredentials.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.gcs;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+
+public class GoogleApplicationDefaultCredentials {
+    private static final Logger logger = LogManager.getLogger(GoogleApplicationDefaultCredentials.class);
+
+    public GoogleCredentials getApplicationDefaultCredentials() {
```

Comment:
Should this be `get()`? These are already `GoogleApplicationDefaultCredentials`.

Another option is to call this class `GoogleApplicationCredentials` and implement `getDefault`.

Reply:
Thanks for reviewing this @dblock, there is a doc in GCP space Google's Application Default Credentials (ADC https://cloud.google.com/docs/authentication/provide-credentials-adc) on how to use this.